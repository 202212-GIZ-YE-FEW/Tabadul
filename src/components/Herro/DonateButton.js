import { HeroButton } from "./hero.style";

const DonateButton = () => {
    return <HeroButton>Donate Now</HeroButton>;
};

export default DonateButton;
