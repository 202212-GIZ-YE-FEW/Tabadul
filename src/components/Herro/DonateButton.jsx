import { onAuthStateChanged } from "firebase/auth";
import { useRouter } from "next/router";
import { useTranslation, withTranslation } from "next-i18next";
import { useEffect, useState } from "react";

import { auth } from "@/utils/firebase";

import { HeroButton } from "./hero.style";

const DonateButton = () => {
    const { t } = useTranslation("hero");
    function translateMe(word) {
        return t(word);
    }
    const route = useRouter();
    const [isAuthenticated, setIsAuthenticated] = useState(false);

    useEffect(() => {
        const Listerner = onAuthStateChanged(auth, async (user) => {
            setIsAuthenticated(Boolean(user));
        });
        return () => {
            Listerner();
        };
    }, []);

    function handleclick() {
        if (isAuthenticated) {
            route.push("/AddItems");
        } else {
            route.push("/Signin");
        }
    }
    return <HeroButton onClick={handleclick}>{t("DonateNow")}</HeroButton>;
};

export default DonateButton;
