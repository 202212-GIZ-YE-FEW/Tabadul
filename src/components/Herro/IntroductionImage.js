import styles from "../../styles/hero.module.css";

const IntroductionImage = () => {
    return (
        <img
            className={styles.image}
            src='/images/hero1.png'
            alt='Introduction'
            width='450'
            height='350'
        />
    );
};

export default IntroductionImage;
