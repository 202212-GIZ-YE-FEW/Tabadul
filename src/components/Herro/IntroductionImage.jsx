import { IntroImage } from "./hero.style";

const IntroductionImage = () => {
    return (
        <IntroImage
            src='/images/hero1.png'
            alt='Introduction'
            width='470'
            height='370'
            data-aos='zoom-in'
        />
    );
};

export default IntroductionImage;
