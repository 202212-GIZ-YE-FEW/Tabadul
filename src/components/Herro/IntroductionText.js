import styles from "../../styles/hero.module.css";

const IntroductionText = () => {
    return (
        <div className={styles.IntroductionText}>
            <h2>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
                eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut
                enim ad minim veniam, quis nostrud exercitation ullamco laboris
                nisi ut aliquip ex ea commodo consequat.
            </h2>
        </div>
    );
};

export default IntroductionText;
