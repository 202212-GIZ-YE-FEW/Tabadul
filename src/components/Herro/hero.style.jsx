import styled from "@emotion/styled";
import Image from "next/image";

export const HeroSection = styled.div`
    background-color: #f1f1f1;
    height: calc(100vh - 50px);
    @media screen and (max-width: 992px) {
        height: auto;
    }
`;
export const Container = styled.div`
    margin-left: auto;
    margin-right: auto;
    padding-left: 15px;
    padding-right: 15px;
    @media (min-width: 768px) {
        width: 750px;
    }
    /* Medium */
    @media (min-width: 992px) {
        width: 970px;
    }
    @media (min-width: 1200px) {
        width: 1170px;
    }
    /* Large */
    @media (min-width: 1500px) {
        width: 1500px;
    }
`;

export const InnerContainer = styled.div`
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
    margin-bottom: 120px;
    padding-top: 104px;
    @media (min-width: 1200px) and (max-width: 1500px) {
        margin-bottom: 40px;
        padding-top: 40px;
    }
    @media screen and (max-width: 1200px) {
        flex-direction: column-reverse;
        padding-top: 0px;
        margin-bottom: 0px;
    }
`;
export const HeroTextContainer = styled.div`
    width: 618px;
    @media (min-width: 1200px) and (max-width: 1500px) {
        width: 518px;
    }
    @media (min-width: 768px) and (max-width: 1200px) {
        width: fit-content;
    }
    @media screen and (max-width: 768px) {
        width: fit-content;
    }
`;
export const HeroText = styled.p`
    font-style: normal;
    font-weight: 400;
    font-size: 32px;
    line-height: 48px;
    color: #000000;
    text-align: left;
    @media (min-width: 1200px) and (max-width: 1500px) {
        width: 470px;
        font-size: 24px;
        line-height: 44px;
        width: 100%;
    }
    @media (min-width: 768px) and (max-width: 1200px) {
        font-size: 28px;
        line-height: 36px;
        text-align: center;
    }
    @media screen and (max-width: 768px) {
        font-size: 22px;
        line-height: 36px;
        text-align: center;
        padding: 5px 30px;
    }
`;
export const HeroButton = styled.button`
    background-color: #33956d;
    color: white;
    width: 292px;
    height: 94px;
    font-style: normal;
    font-weight: 700;
    font-size: 36px;
    line-height: 54px;
    text-align: center;
    border: none;
    border-radius: 20px;
    cursor: pointer;
    border-radius: 54px;
    margin-left: 100px;
    margin-top: 30px;
    &:hover {
        background-color: #267548;
    }
    @media screen and (max-width: 1500px) {
        width: 190px;
        height: 60px;
        font-size: 24px;
        margin-left: 60px;
    }
    @media (min-width: 768px) and (max-width: 1200px) {
        display: flex;
        align-items: center;
        justify-content: center;
        margin: auto;
    }
    @media screen and (max-width: 768px) {
        font-size: 24px;
        width: 184px;
        height: 59px;
        margin-left: 0px;
        margin-top: 50px;
        margin: auto;
        display: flex;
        align-items: center;
        justify-content: center;
    }
`;

export const IntroImage = styled(Image)`
    width: 780px;
    height: 629px;
    max-width: 100%;
    object-fit: contain;
    @media (min-width: 1200px) and (max-width: 1500px) {
        width: 500px;
        height: 500px;
    }
    @media (min-width: 768px) and (max-width: 1200px) {
        width: 500px;
        height: 500px;
    }
    @media screen and (max-width: 768px) {
        width: 336px;
        height: 336px;
    }
    @media screen and (max-width: 375px) {
        width: 280px;
        height: 280px;
    }
`;

export const LeftSideDiv = styled.div`
    flex: 1;
`;

export const Line = styled.div`
    position: relative;
    left: 50%;
    transform: translateX(-50%);
    width: 100%;
    height: 1px;
    background-color: #33956d;
    bottom: 50%;
    margin-bottom: 20px;
    @media screen and (max-width: 1200px) {
        display: none;
    }
`;
