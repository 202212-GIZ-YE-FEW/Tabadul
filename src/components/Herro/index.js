import DonateButton from "./DonateButton";
import {
    Container,
    HeroSection,
    InnerContainer,
    LeftSideDiv,
    Line,
} from "./hero.style";
import IntroductionImage from "./IntroductionImage";
import IntroductionText from "./IntroductionText";

function Hero() {
    return (
        <HeroSection>
            <Container>
                <InnerContainer>
                    <LeftSideDiv>
                        <IntroductionText />
                        <DonateButton />
                    </LeftSideDiv>
                    <IntroductionImage />
                </InnerContainer>
            </Container>
            <Line />
        </HeroSection>
    );
}

export default Hero;
