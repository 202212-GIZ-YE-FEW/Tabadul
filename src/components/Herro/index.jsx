import DonateButton from "./DonateButton";
import {
    Container,
    HeroSection,
    InnerContainer,
    LeftSideDiv,
    Line,
} from "./hero.style";
import IntroductionImage from "./IntroductionImage";
import IntroductionText from "./IntroductionText";
function Hero() {
    return (
        <HeroSection>
            <Container>
                <InnerContainer>
                    <LeftSideDiv data-aos='fade-right'>
                        <IntroductionText />
                        <DonateButton />
                    </LeftSideDiv>
                    <IntroductionImage />
                </InnerContainer>
                <Line />
            </Container>
        </HeroSection>
    );
}

export default Hero;
