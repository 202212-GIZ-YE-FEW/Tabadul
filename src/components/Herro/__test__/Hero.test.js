import renderer from "react-test-renderer";

import Herro from "..";

jest.mock("react-i18next", () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

it("renders correctly", () => {
    const tree = renderer.create(<Herro />).toJSON();
    expect(tree).toMatchSnapshot();
});

jest.mock("next/router", () => ({
    useRouter() {
        return {
            route: "/",
            pathname: "",
            query: "",
            asPath: ",",
        };
    },
}));
