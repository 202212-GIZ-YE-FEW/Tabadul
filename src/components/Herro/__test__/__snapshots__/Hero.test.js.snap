// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-n708sp"
>
  <div
    className="css-w5ihqc"
  >
    <div
      className="css-151p8ll"
    >
      <div
        className="css-eie7o"
      >
        <div
          className="css-1j4g1g"
        >
          <p
            className="css-y41u20"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
          </p>
        </div>
        <button
          className="css-thbdpf"
        >
          Donate Now
        </button>
      </div>
      <img
        alt="Introduction"
        className="css-9japz0"
        data-nimg="1"
        decoding="async"
        height={370}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2Fhero1.png&w=1080&q=75"
        srcSet="/_next/image?url=%2Fimages%2Fhero1.png&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fhero1.png&w=1080&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={470}
      />
    </div>
    <div
      className="css-1z0br83"
    />
  </div>
</div>
`;
